### DISCUSSION TITLE
view:unsplit semantics

### DISCUSSION THREAD

**Discussion by eric1234** [orbitalquark/textadept#287] [Q&A] [answered]
I'm trying to understand how view:unsplit works. Lets say I have the following layout:

- View 1
- Split Table
  - View 2
  - View 3
  
Basically I have hit `<ctrl>-<alt>-v` to split in half. Then on the right half hit `<ctrl>-<alt>-v` to split again giving me a total of three views. For each of these views how would I remove that view? My testing shows:

`_VIEW[1]:unsplit()` - this will close both view 2 and 3 leaving just view 1
`_VIEW[2]:unsplit()` - this will close view 3 leaving view 1 and 2
`_VIEW[3]:unsplit()` - this closes view 2 leaving 1 and 3

The general logic that seems to be merging is whatever other pane is paired with the view I ask it to unsplit is what is removed. That other pane may just be a single view or it could be a whole tree of views.

In the above example is there a way to remove view 1 while leaving 2 and 3 (without just doing something like removing all views and rebuilding the layout without view 1)?

**Comment by orbitalquark** [comment, marked as answer]
No, there is no way to remove just view 1, leaving views 2 and 3 in your scenario. Splitting a view replaces it with a new "node" that contains the original view and a new view. This results in something like a tree structure. Unsplitting a given view only removes the other "node" from the tree.

**Reply by eric1234** [reply]
Thanks for the confirmation. I ended up faking it.

If I have say 5 columns and I want to remove the 2nd column I have the 2nd column open the buffer of the 3rd column, the 3rd column open the buffer of the 4th column, the 4th column open the buffer of the 5th column and finally unsplit the 4th view (which closes the 5th view). End result is the effect I want (it looks like I closed my 2nd column) even if under the covers it's doing a bit of a dance.
